### PR TITLE
Adding version check

### PR DESCRIPTION
**Pull request by leviathanch**
This adds a missing version check, making FreeCAD compile with Qt6

### CODE REVIEW

**Review by davesrocketshop — APPROVED** [review]

File: src/Mod/Material/App/MaterialLoader.cpp [hunk 1]
```diff
@@ -29,6 +29,10 @@
 #include <QList>
 #include <QMetaType>
 
+#if QT_VERSION >= QT_VERSION_CHECK(6,0,0)
+#include <QtCore5Compat/QRegExp>
```
Comment (chennes) [inline src/Mod/Material/App/MaterialLoader.cpp:33]: Can the code be modified to use `QRegularExpression` instead? `QRegExp` is very old.

Comment (davesrocketshop) [reply src/Mod/Material/App/MaterialLoader.cpp:33]: Yes. This should be a direct substitution in this code.

Comment (chennes) [reply src/Mod/Material/App/MaterialLoader.cpp:33]: Are you doing that, or should @leviathanch modify this PR to take that course of action instead?

Comment (davesrocketshop) [reply src/Mod/Material/App/MaterialLoader.cpp:33]: I can do it but that will come in a future PR. This should be fine in the interim.

**Pull request merged**